src/IRI.php

```diff
@@ -547,9 +556,6 @@ protected function remove_iunreserved_percent_encoded($match)
                 // Start position
                 $start = $i;
 
-                // By default we are valid
-                $valid = true;
```

Comment:
This is still necessary, otherwise invalid parse will make rest of the string invalid.

Reply:
Sorry, you are right. This should be fixed now.